Manipulation/ improve 3D perception

Fernando94654 opened the pull request:
Optimization in filter pointCloud process
Improved local processing times:
Without improvements:
<img width="1367" height="630" alt="Screenshot from 2025-09-15 09-32-09" src="https://github.com/user-attachments/assets/22f1750e-f582-463f-8516-3cb9b2a4bf1a" />

With preallocation memory:
<img width="1513" height="622" alt="Screenshot from 2025-09-15 09-53-06" src="https://github.com/user-attachments/assets/f3385e5d-0423-4bf8-988b-444da482d919" />

With parallel filtering and simplifiator_factor = 3:
[
<img width="1544" height="564" alt="Screenshot from 2025-09-15 13-00-52" src="https://github.com/user-attachments/assets/111ff6f8-7a19-4c25-9589-211daf0016cd" />

Fernando94654 committed:
optimization in filter pointCloud process

Fernando94654 committed:
filter_cloud name correction

JLDominguezM commented on manipulation/packages/perception_3d/src/down_sample_pc.cpp:
```diff
@@ -43,6 +44,7 @@ class DownSamplePointCloud : public rclcpp::Node {
   float medium_radius = 2.5f; // 2.5m
   float sqr_small_rad;
   float sqr_med_rad;
+  int simplifiator_factor = 3;
```
also change the name to simplificator_factor

Fernando94654 committed:
rename simplificator_factor

Fernando94654 committed:
merge main
Merge remote-tracking branch 'origin/main' into manipulation/improve-3Dperception

JLDominguezM reviewed (approved):
LGTM 🚀
